From 6df628bedc3f0e524cf2b8afe42378d3da9836db Mon Sep 17 00:00:00 2001
From: Amithash Prasad <amithash@meta.com>
Date: Thu, 15 Oct 2026 14:25:59 -0700
Subject: [PATCH] platform-mc: Do no emit signals on updates to UpdatedTime

Since UpdatedTime is updated on every update irrespective of whether the
value has changed, it is not useful to emit a signal for it. It can
in fact overwhelm any signal listeners and does not provide any useful
information to them.

Change-Id: I275846f17755461cd72f873b2ccf2c01f44cb930
Signed-off-by: Amithash Prasad <amithash@meta.com>
---
 platform-mc/numeric_sensor.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/platform-mc/numeric_sensor.cpp b/platform-mc/numeric_sensor.cpp
index 2674ce7..db557dc 100644
--- a/platform-mc/numeric_sensor.cpp
+++ b/platform-mc/numeric_sensor.cpp
@@ -743,11 +743,14 @@ void NumericSensor::updateReading(bool available, bool functional, double value)
             if (!useMetricInterface)
             {
                 valueIntf->value(newValue);
-                // Refresh timestamp only when the new value is valid.
+                // Refresh timestamp only when the new value is valid
+                // and considering this is done irrespecitve of the
+                // value change, do not emit signals.
                 if (std::isfinite(newValue))
                 {
                     valueIntf->updatedTime(
-                        pldm::utils::getCurrentSystemTimeInMicroseconds());
+                        pldm::utils::getCurrentSystemTimeInMicroseconds(),
+                        true);
                 }
                 updateThresholds();
             }
-- 
2.52.0

